java/serve/src/main/java/io/ray/serve/api/Serve.java

```diff
@@ -1,41 +1,249 @@
 package io.ray.serve.api;
 
-import com.google.common.base.Preconditions;
+import io.ray.api.ActorHandle;
 import io.ray.api.BaseActorHandle;
+import io.ray.api.PyActorHandle;
 import io.ray.api.Ray;
-import io.ray.serve.Constants;
-import io.ray.serve.RayServeException;
-import io.ray.serve.ReplicaContext;
+import io.ray.api.call.PyActorCreator;
+import io.ray.api.exception.RayActorException;
+import io.ray.api.exception.RayTimeoutException;
+import io.ray.api.function.PyActorClass;
+import io.ray.api.function.PyActorMethod;
+import io.ray.api.options.ActorLifetime;
+import io.ray.serve.common.Constants;
+import io.ray.serve.deployment.Deployment;
+import io.ray.serve.deployment.DeploymentCreator;
+import io.ray.serve.deployment.DeploymentRoute;
+import io.ray.serve.exception.RayServeException;
+import io.ray.serve.generated.ActorNameList;
+import io.ray.serve.proxy.ProxyActor;
+import io.ray.serve.replica.ReplicaContext;
+import io.ray.serve.util.CollectionUtil;
+import io.ray.serve.util.CommonUtil;
 import io.ray.serve.util.LogUtil;
+import io.ray.serve.util.ServeProtoUtil;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Optional;
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** Ray Serve global API. TODO: will be riched in the Java SDK/API PR. */
 public class Serve {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(Serve.class);
+
   private static ReplicaContext INTERNAL_REPLICA_CONTEXT;
 
-  private static Client GLOBAL_CLIENT;
+  private static ServeControllerClient GLOBAL_CLIENT;
+
+  private static Pattern ANONYMOUS_NAMESPACE_PATTERN =
```

Comment:
#25575 will ensure that all Serve actors start only in the `"serve"` namespace. Could we add that change to this PR?

Reply:
OK. Simon mentioned this optimization before. Let's modify it synchronously.